src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -630,7 +630,7 @@ var ReactCompositeComponentMixin = {
     if (this._pendingElement != null) {
       ReactReconciler.receiveComponent(
         this,
-        this._pendingElement || this._currentElement,
+        this._pendingElement,
```

Comment:
when `this._pendingElement` is undefined, `this._pendingElement != null` is false value.